Enable all text overlays for items in hotbar by default.

ekgame opened the pull request:
<table>
<tr>
	<td>Before
	<td>After
<tr>
	<td><img src="https://user-images.githubusercontent.com/1730245/188880831-043f6586-871f-4be7-82ca-fd9773d81895.png">
	<td><img src="https://user-images.githubusercontent.com/1730245/188880774-9fe646ab-c764-4f6b-ac19-f7faf6029fa2.png">
</table>

There is a setting for enabling/disabling text overlays in hotbars, however it is disregarded for most items. This pull request makes all items render in the hotbar as expected when the setting is enabled.

Incompleteusern commented on src/main/java/com/wynntils/modules/utilities/managers/PouchHotkeyManager.java:
```diff
@@ -31,21 +32,54 @@ public static void onEmeraldHotkeyPress() {
+
+                // At this point, we have either multiple pouches with zero emeralds, or multiple pouches but only one with a non-zero balance
+                // Check to make sure we don't have a bunch of zero balances - if we do, notify user
+                System.out.println(emeraldPouches.values().stream().mapToInt(Integer::intValue).sum());
```
Seems to be left over debug, the condition is also the same as alreadyHasNonEmpty

Incompleteusern commented on src/main/java/com/wynntils/modules/utilities/managers/PouchHotkeyManager.java:
```diff
@@ -31,21 +32,54 @@ public static void onEmeraldHotkeyPress() {
+                }
+
+                // Now, we know we have 1 used pouch and 1+ empty pouches - just open the used one
+                List<Integer> pouchSlotList = new ArrayList<>();
```
This could probably be merged with the above for loop by saving the index of the non empty pouch

Incompleteusern reviewed (changes requested):
LGTM, but a few things

Incompleteusern commented on src/main/java/com/wynntils/modules/utilities/managers/PouchHotkeyManager.java:
```diff
@@ -46,32 +44,28 @@ public static void onEmeraldHotkeyPress() {
                 // At this point, we have either multiple pouches with zero emeralds, or multiple pouches but only one with a non-zero balance
                 // Check to make sure we don't have a bunch of zero balances - if we do, notify user
-                System.out.println(emeraldPouches.values().stream().mapToInt(Integer::intValue).sum());
                 if (emeraldPouches.values().stream().mapToInt(Integer::intValue).sum() < 1) {
```
This condition is the same as alreadyHasNonEmpty, probably could merge and rename the boolean to fit

ekgame committed:
Enable all text overlays for items in hotbar by default.

The setting that handles if the text overlays should be rendered is handled in ItemTextOverlayFeature#onRenderHotbarSlot.

kristofbolyai commented:
> There is a setting for enabling/disabling text overlays in hotbars, however it is disregarded for most items.

What do you mean?

kristofbolyai commented:
Oh, I see. What I don't get is how this change does not ignore the "hotbar text overlays" setting? 

Does the fix make it so everything  gets rendered unless specified separately otherwise or the setting is turned off?

ekgame commented:
Whether or not text overlays should be rendered in the hotbar is handled here: https://github.com/Wynntils/Artemis/blob/f441312ae88d1d99ffc2764f4938f442bbce8fe9/common/src/main/java/com/wynntils/features/user/inventory/ItemTextOverlayFeature.java#L84 

It can be turned on/off in the settings: 
![paveikslas](https://user-images.githubusercontent.com/1730245/188886195-53b80733-aa65-40f1-941c-321816a6f839.png)


However, the `isHotbarText` method is an override - if it returns false, it disregards the setting and refuses to render the overlays as expected.

kristofbolyai commented:
Nice! Thanks.
